test cases/common/205 argumnet type/meson.build

```diff
@@ -0,0 +1,25 @@
+project(
+  'argument type',
+  ['c'],
+)
+
+cc = meson.get_compiler('c')
+
+if ['gcc', 'lcc', 'clang'].contains(cc.get_id())
+  expected = 'gcc'
+elif cc.get_id() == 'msvc'
+  expected = 'msvc'
+elif cc.get_id() == 'icc'
+  if host_machine.system() == 'windows'
+    expected = 'msvc'
+  else
+    expected = 'gcc'
+  endif
+else
+  # It's possible that other compilers end up here that shouldn't
```

Comment:
Probably the test should fail here: It indicates a compiler the test case doesn't know about?